theMoth.py

```diff
@@ -19,15 +19,46 @@ async def on_message(message):
         await message.channel.send('Hello!')
 
     if message.content.startswith('?help'):
-        await message.channel.send('Supported commands:\n`?sooth` - draws a random sooth card\n`?roll` - rolls dice, e.g. `?roll 2d6` or `?roll d20 +5`\n\nWith no arguments, `?roll` will roll a mundane die from Invisible Sun, and with a +[number] argument it will roll a mundane die plus that many magic dice, e.g. `?roll +3`')
+        await message.channel.send('Supported commands:'
+            '\n`?sooth` - draws a random sooth card'
+            '\n`?char` - generates a Suns Apart character'
+            '\n`?roll` - rolls dice, e.g. `?roll 2d6` or `?roll d20 +5`'
+            '\n- With no arguments, `?roll` will roll a mundane die from Invisible Sun, and with a +[number] argument it will roll a mundane die plus that many magic dice, e.g. `?roll +3`'
+            '\n`?save` - rolls a d20. Optionally provide an argument to compare against')
 
     if message.content.startswith('?sooth'):
         # generate a random number between 1 and 60, with leading zeros if needed
         cardNum = str(random.randint(1,60)).zfill(2)
         # post the link to the card image matching that number (Discord should auto-embed the image)
         await message.channel.send("https://app.invisiblesunrpg.com/wpsite/wp-content/uploads/2018/04/"+cardNum+".png")
         # post the link to the details page for that card. The <> wrapping on the URL prevents Discord from embedding a link preview (looks cleaner that way)
-        await message.channel.send("Details: <https://app.invisiblesunrpg.com/soothdeck/card-"+cardNum+"/>")
+        return await message.channel.send("Details: <https://app.invisiblesunrpg.com/soothdeck/card-"+cardNum+"/>")
+
+    if message.content.startswith('?char'):
+        # Using Suns Apart flux:
+        flux = lambda x: ['', '*', '!'][len(x) - len(set(x))]
+        ret = ['```']
+        for s in ['CER', 'QUA', 'SOR']:
+            d = [random.randrange(1, 7) for _ in range(3)]

```

Comment:
As I mentioned in a latter comment, `randrange()` may be a bit confusing/unnatural to read and can lead to inconsistencies as in line 50. Maybe we should use `randint()` since looking at `randint(1, 6)` makes it more clear what we're trying to generate? Not sure if there are other implications of using that over `randrange()`,

Reply:
Nope, seems like it should be fine.